File: _performance-golf/pg-ai/_gemini-cli/api-examples/go/text_generation_test.go
package examples

import (
	"testing"
)

func TestTextGenTextOnlyPrompt(t *testing.T) {
	_, err := TextGenTextOnlyPrompt()
	if err != nil {
		t.Errorf("TextGenTextOnlyPrompt returned an error.")
	}
}

func TestTextGenTextOnlyPromptStreaming(t *testing.T) {
	err := TextGenTextOnlyPromptStreaming()
	if err != nil {
		t.Errorf("TextGenTextOnlyPromptStreaming returned an error.")
	}
}

func TestTextGenMultimodalOneImagePrompt(t *testing.T) {
	_, err := TextGenMultimodalOneImagePrompt()
	if err != nil {
		t.Errorf("TextGenMultimodalOneImagePrompt returned an error.")
	}
}

func TestTextGenMultimodalOneImagePromptStreaming(t *testing.T) {
	err := TextGenMultimodalOneImagePromptStreaming()
	if err != nil {
		t.Errorf("TextGenMultimodalOneImagePromptStreaming returned an error.")
	}
}

func TestTextGenMultimodalMultiImagePrompt(t *testing.T) {
	_, err := TextGenMultimodalMultiImagePrompt()
	if err != nil {
		t.Errorf("TextGenMultimodalMultiImagePrompt returned an error.")
	}
}

func TestTextGenMultimodalMultiImagePromptStreaming(t *testing.T) {
	err := TextGenMultimodalMultiImagePromptStreaming()
	if err != nil {
		t.Errorf("TextGenMultimodalMultiImagePromptStreaming returned an error.")
	}
}

func TestTextGenMultimodalAudio(t *testing.T) {
	_, err := TextGenMultimodalAudio()
	if err != nil {
		t.Errorf("TextGenMultimodalAudio returned an error.")
	}
}

func TestTextGenMultimodalAudioStreaming(t *testing.T) {
	err := TextGenMultimodalAudioStreaming()
	if err != nil {
		t.Errorf("TextGenMultimodalAudioStreaming returned an error.")
	}
}

func TestTextGenMultimodalVideoPrompt(t *testing.T) {
	_, err := TextGenMultimodalVideoPrompt()
	if err != nil {
		t.Errorf("TextGenMultimodalVideoPrompt returned an error.")
	}
}

func TestTextGenMultimodalVideoPromptStreaming(t *testing.T) {
	err := TextGenMultimodalVideoPromptStreaming()
	if err != nil {
		t.Errorf("TextGenMultimodalVideoPromptStreaming returned an error.")
	}
}

func TestTextGenMultimodalPdf(t *testing.T) {
	_, err := TextGenMultimodalPdf()
	if err != nil {
		t.Errorf("TextGenMultimodalPdf returned an error.")
	}
}

func TestTextGenMultimodalPdfStreaming(t *testing.T) {
	err := TextGenMultimodalPdfStreaming()
	if err != nil {
		t.Errorf("TextGenMultimodalPdfStreaming returned an error.")
	}
}

File: _performance-golf/pg-ai/_gemini-cli/api-examples/javascript/text_generation.test.js
/**
 * @license
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import assert from "node:assert";
import { test, describe } from "node:test";
import {
  textGenTextOnlyPrompt,
  textGenTextOnlyPromptStreaming,
  textGenMultimodalOneImagePrompt,
  textGenMultimodalOneImagePromptStreaming,
  textGenMultimodalMultiImagePrompt,
  textGenMultimodalMultiImagePromptStreaming,
  textGenMultimodalAudio,
  textGenMultimodalAudioStreaming,
  textGenMultimodalVideoPrompt,
  textGenMultimodalVideoPromptStreaming,
  textGenMultimodalPdf,
  textGenMultimodalPdfStreaming,
} from "./text_generation.js";

describe("text_generation", () => {
  test("textGenTextOnlyPrompt", async () => {
    const text = await textGenTextOnlyPrompt();
    assert.ok(text.length > 0);
  });

  test("textGenTextOnlyPromptStreaming", async () => {
    const text = await textGenTextOnlyPromptStreaming();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalOneImagePrompt", async () => {
    const text = await textGenMultimodalOneImagePrompt();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalOneImagePromptStreaming", async () => {
    const text = await textGenMultimodalOneImagePromptStreaming();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalMultiImagePrompt", async () => {
    const text = await textGenMultimodalMultiImagePrompt();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalMultiImagePromptStreaming", async () => {
    const text = await textGenMultimodalMultiImagePromptStreaming();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalAudio", async () => {
    const text = await textGenMultimodalAudio();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalAudioStreaming", async () => {
    const text = await textGenMultimodalAudioStreaming();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalVideoPrompt", async () => {
    const text = await textGenMultimodalVideoPrompt();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalVideoPromptStreaming", async () => {
    const text = await textGenMultimodalVideoPromptStreaming();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalPdf", async () => {
    const text = await textGenMultimodalPdf();
    assert.ok(text.length > 0);
  });

  test("textGenMultimodalPdfStreaming", async () => {
    const text = await textGenMultimodalPdfStreaming();
    assert.ok(text.length > 0);
  });
});
